### PR TITLE
Add decision for Secret key managment solution

### PR DESCRIPTION
**Pull request by TheBigLee**
## Summary

This PR adds the decision for the secretkey management solution that we want to implement for AppCat
## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `change`, `decision`, `requirement/quality`, `requirement/functional`, `dependency`
      as they show up in the changelog
- [ ] Link this PR to related issues if applicable.

### CODE REVIEW

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+We therefore need to dig deeper to figure out which solution is the best for us.
+
+=== Analysis Vault
+
+Vault is the most popular and widespread solution on the market. It is probably the most feature-complete solution on the market as of now.
+However, it recently changed it's license to https://www.hashicorp.com/bsl[BSL]. This unfortunately limits how we can offer Vault to our customers.
+
+Advantages::
+
+* The de facto standard in the industry
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc:29]:
```suggestion
* The de facto industry standard
```

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+* https://www.vaultproject.io[Hashicorp Vault]
+* https://openbao.org[OpenBao]
+* https://github.com/Infisical/infisical[Infisical]
+
+All listed solutions meet the requirements.
+We therefore need to dig deeper to figure out which solution is the best for us.
+
+=== Analysis Vault
+
+Vault is the most popular and widespread solution on the market. It is probably the most feature-complete solution on the market as of now.
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc:24]: You repeat `on the market` twice here.

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+Advantages::
+
+* Open-Source
+* Self-Hostable
+
+
+Disadvantages::
+
+* Many key features are not open-source and require the paid version:
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc:72]: Checking their page it also looks like they don't have any PKI at all... this alone is probably already a k.o. criterium.

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+** RBAC
+** Secret Rotation
+** SAML SSO
+** IP Allowlisting
+
+* Not compatible with the Vault API
+
+== Decision
+
+We use OpenBao
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc:83]: While I agree with going with OpenBao, it currently doesn't even have a release or a helm chart...

https://github.com/openbao/openbao

We should probably wait a while for this fork to get settled before actually using it.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Some small things I found. I generally agree with the decision though.

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc (hunk 4 above)
Comment (wejdross) [reply docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc:83]: https://openbao.org/docs/platform/k8s/helm/

they expose link to `helm for kubernetes` but it throws 404 :D

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+* Open-Source
+* Self-Hostable
+* Free of charge
+* Compatible with the Vault API
+* Backed by the Linux Foundation
+
+Disadvantages::
+
+* Might deviate from Vault in the future and not be compatible anymore
```
Comment (zugao) [inline docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc:55]: I wouldn't say per se this is a disadvantage as we treat OpenBao as a different tool than Vault.

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+* Not compatible with the Vault API
+
+== Decision
+
+We use OpenBao
+
+== Rationale
+
+OpenBao is the only true Open-Source solution and thanks to the compatibility with the Vault API is also very well established in the industry.
+Furthermore it is maintained by the Linux Foundation, which gives us high confindence, that the project will not be abandonded quickly.
```
Comment (zugao) [inline docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc:88]: I guess this is the sell for me :) So I agree with the decision.

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc (hunk 4 above)
Comment (TheBigLee) [reply docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc:83]: Yes I agree. They haven't ported the helm chart yet. We could probably use the official vault helm chart use the OpenBao image until they have their own chart ready.

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc (hunk 2 above)
Comment (TheBigLee) [reply docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc]: Fixed

File: docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc (hunk 4 above)
Comment (TheBigLee) [reply docs/modules/ROOT/pages/explanations/decisions/secret-pki-mgmt.adoc:84]: We could use the official vault helm chart and override the image until they have their own chart.

**Review by Kidswiss — APPROVED** [review]

**Review by zugao — APPROVED** [review]

**Pull request merged**